### PR TITLE
10.0 framework

### PR DESCRIPTION
**Pull request by StefanRijnhart**

### CODE REVIEW

**Comment by StefanRijnhart** [comment]
If you are interested, here is a preview of the analysis generated with this change:
https://github.com/StefanRijnhart/OpenUpgrade/commit/799151b7d26f3dbddf63b55c29e483a20ee23e10
https://github.com/StefanRijnhart/OpenUpgrade/commit/e71012fb5af55e3da23c12e70b6ed16ff21f62a8

File: odoo/addons/openupgrade_records/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# coding: utf-8
+# Copyright 2011-2015 Therp BV <https://therp.nl>
+# Copyright 2016 Opener B.V. <https://opener.am>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+{
+    'name': 'OpenUpgrade Records',
+    'version': '0.3',
```
Comment (pedrobaeza) [inline odoo/addons/openupgrade_records/__manifest__.py]: Should we use 10.0.1.0.0?

File: odoo/addons/openupgrade_records/lib/compare.py [hunk 2]
```diff
@@ -0,0 +1,277 @@
+# -*- coding: utf-8 -*-
+##############################################################################
```
Comment (pedrobaeza) [inline odoo/addons/openupgrade_records/lib/compare.py]: Use short headers.

File: odoo/openupgrade/deferred_100.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# coding: utf-8
```
Comment (pedrobaeza) [inline odoo/openupgrade/deferred_100.py:1]: The file name seems awkward.

File: odoo/openupgrade/doc/source/modules50-60.rst [hunk 4]
```diff
@@ -0,0 +1,342 @@
+Module coverage 5.0 -> 6.0
```
Comment (pedrobaeza) [inline odoo/openupgrade/doc/source/modules50-60.rst:1]: We should remove these files, as they are catched from corresponding branches.

File: requirements.txt [hunk 5]
```diff
@@ -40,3 +40,4 @@ vobject==0.6.6
 wsgiref==0.1.2
 XlsxWriter==0.7.7
 xlwt==0.7.5
+openupgradelib>=1.1.0
```
Comment (pedrobaeza) [inline requirements.txt]: We should raise this to the new version that includes the imports.

File: scripts/compare_noupdate_xml_records.py [hunk 6]
```diff
@@ -0,0 +1,213 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+##############################################################################
```
Comment (pedrobaeza) [inline scripts/compare_noupdate_xml_records.py]: Short header

File: scripts/migrate.py [hunk 7]
```diff
@@ -0,0 +1,428 @@
...
+            ('pg_dump --format=custom --no-password %(db_old)s ' +
+             '| pg_restore --no-password --dbname=%(db_new)s') %
+            {'db_old': db_old, 'db_new': db_new}
+        )
+
+        if password_set:
+            del os.environ['PGPASSWORD']
+    return db_new
+
+migrations = {
```
Comment (pedrobaeza) [inline scripts/migrate.py:61]: Don't duplicate variable

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: odoo/addons/openupgrade_records/__manifest__.py (hunk 1 above)
Comment (StefanRijnhart) [reply odoo/addons/openupgrade_records/__manifest__.py]: Done

File: odoo/addons/openupgrade_records/lib/compare.py (hunk 2 above)
Comment (StefanRijnhart) [reply odoo/addons/openupgrade_records/lib/compare.py]: Done

File: odoo/openupgrade/deferred_100.py (hunk 3 above)
Comment (StefanRijnhart) [reply odoo/openupgrade/deferred_100.py:1]: Just a matter of getting used to ;-) It _is_ consistent with previous iterations.

File: requirements.txt (hunk 5 above)
Comment (StefanRijnhart) [reply requirements.txt]: I tagged 1.2.0 on Github, but I think only board members can push it to Pypi, right?

File: scripts/compare_noupdate_xml_records.py (hunk 6 above)
Comment (StefanRijnhart) [reply scripts/compare_noupdate_xml_records.py]: Done

File: scripts/migrate.py (hunk 7 above)
Comment (StefanRijnhart) [reply scripts/migrate.py:61]: Done.

File: odoo/openupgrade/doc/source/modules50-60.rst (hunk 4 above)
Comment (StefanRijnhart) [reply odoo/openupgrade/doc/source/modules50-60.rst:1]: I kept the files as placeholders so as not to break the documentation build, containing a link to the published version.

File: requirements.txt (hunk 5 above)
Comment (pedrobaeza) [reply requirements.txt]: The publication to PyPi is made automatically when tagging a version, so you have implicit permissions to do push :wink:

File: odoo/openupgrade/doc/source/modules50-60.rst (hunk 4 above)
Comment (pedrobaeza) [reply odoo/openupgrade/doc/source/modules50-60.rst:1]: But this is correctly joined by documentation script, so no need to this placeholder I think

Comment (StefanRijnhart) [reply odoo/openupgrade/doc/source/modules50-60.rst:1]: But I have to keep the references to the files in the referring RST file, and without the files the docs won't build locally.

File: requirements.txt (hunk 5 above)
Comment (StefanRijnhart) [reply requirements.txt]: Nice! I was going through the Travis logs for a hint on this but I did not find anything. But you are right https://pypi.python.org/pypi/openupgradelib/1.2.0

File: odoo/openupgrade/doc/source/modules50-60.rst (hunk 4 above)
Comment (pedrobaeza) [reply odoo/openupgrade/doc/source/modules50-60.rst:1]: OK, let's keep them

File: requirements.txt (hunk 5 above)
Comment (pedrobaeza) [reply requirements.txt]: This is the configuration for doing it: https://github.com/OCA/openupgradelib/blob/master/.travis.yml#L26-L32

Comment (pedrobaeza) [reply requirements.txt]: And this one set the rest of the data: https://github.com/OCA/openupgradelib/blob/master/setup.py

**Review by pedrobaeza — APPROVED** [review]

**Comment by StefanRijnhart** [comment]
@OCA/openupgrade-maintainers any takers?

**Comment by pedrobaeza** [comment]
We can merge this semi-optimistically :wink:

**Comment by StefanRijnhart** [comment]
Thanks! Here is the analysis: https://github.com/OCA/OpenUpgrade/pull/644

**Pull request merged**